/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import "mesos.proto";
import "containerizer/containerizer.proto";

package mesos.containerizer.examples;

enum FutureStatus {
  FUTURE_PENDING = 0;
  FUTURE_READY = 1;
  FUTURE_FAILED = 2;
  FUTURE_DISCARDED = 3;
}

message FutureMessage
{
  required FutureStatus status = 1;
  optional string message = 2;
}

message LaunchResult
{
  required FutureMessage future = 1;
}

message UsageResult
{
  required FutureMessage future = 1;
  optional ResourceStatistics result = 3;
}

message UpdateResult
{
  required FutureMessage future = 1;
}

message WaitResult
{
  required FutureMessage future = 1;
  optional Termination result = 2;
}

message RecoverRequest {}

message RecoverResult
{
  required FutureMessage future = 1;
}

message DestroyResult {}

message ContainersRequest {}

message ContainersResult
{
  required FutureMessage future = 1;
  optional Containers result = 2;
}
